### PR TITLE
Deprecates Cross Application Tracing (CAT).

### PR DESCRIPTION
**Pull request by michaelgoin**
## Proposed Release Notes

* **Deprecation Warning**: Cross Application Tracing (CAT) has been deprecated and will be removed in a future major release. For applications that explicitly disable Distributed Tracing (DT) to leverage CAT, we recommend migrating to DT to avoid loss of cross-service visibility.
  * Disables CAT by default. You must explicitly enable CAT along with turning off DT.
  * Adds a deprecation warning when CAT is enabled and active (DT disabled).

## Links

* https://github.com/newrelic/node-newrelic/issues/895

## Details

Logs the following warning once, after applying server configuration:

```
[Deprecation Warning]: Cross Application Tracing (CAT) has been deprecated and will be removed in a future major release. CAT has been replaced by Distributed Tracing (DT). Enable DT by setting distributed_tracing: { enabled: true }.
```

### CODE REVIEW

**Comment by michaelgoin** [comment]
While poking through tests found an existing bug that really only should impact our message broker instrumentation but will potentially be more likely to happen with landing CAT as default disabled (as opposed to overriden by DT).

Added a separate issue to fix: https://github.com/newrelic/node-newrelic/issues/900

File: lib/config/default.js [hunk 1]
```diff
@@ -871,10 +871,12 @@ exports.config = () => ({
   /**
    * Controls the use of cross-application tracing.
    *
-   * @property {bool} [enabled=true]
-   *  Enables tracing transactions across multiple applications. Default is `true`.
+   * @property {bool} [enabled=false]
```
Comment (bizob2828) [inline lib/config/default.js:874]: unrelated but this is invalid jsdoc type. should be `boolean`

File: lib/config/index.js [hunk 2]
```diff
@@ -1580,6 +1582,19 @@ Config.prototype.getAggregatorConfig = function getAggregatorConfig(method) {
   }
 }
 
+Config.prototype._warnDeprecations = function _warnDeprecations() {
```
Comment (bizob2828) [inline lib/config/index.js:1585]: i can't seem to comment inline but the two linting warnings above didn't get auto fixed, mind manually fixing?

**Review by bizob2828 — APPROVED** [review]
a few unrelated things I noticed when looking at the diff in context. i'll defer to you to decide if you want to fix the minor issues

File: lib/config/index.js (hunk 2 above)
Comment (michaelgoin) [reply lib/config/index.js:1585]: yeah, code areas that were not updated at all don't allow for inline comments.

happy to take a quick look at the warnings.

Comment (michaelgoin) [reply lib/config/index.js:1586]: @bizob2828 pushed up a commit to clean up the extra things you commented on.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**